xdsl/dialects/experimental/stencil.py

```diff
@@ -481,16 +481,13 @@ class ApplyOp(IRDLOperation):
     def get(
         args: Sequence[SSAValue] | Sequence[Operation],
         body: Block,
+        result_type: AnyFloat,
+        result_rank: int,
```

Comment:
Why not directly a TempType[...] here?

Reply:
We can have multiple results now too. They would all have the same shape but can hold different element types.

I would go for either making `result_type` a Sequence of types, or completely replace `result_type`, `result_rank` and `result_count` by a `result_types` Sequence of TempType[...].

Note that this is nitpicking, especially for experimental! Happy to merge as-is in the meantime

Reply:
Thanks, good suggestions :) I have updated the code based on this